### ISSUE TITLE
Would el-get possible for this project?

### ISSUE THREAD

**Issue by cies** [chrisdone/emacs-haskell-config#6]
This is not much of an issue with this project, more an anecdote on how I (not) use it.

I would love to use a fully functional Haskell "IDE", and for this matter I have switched from Vi to Emacs+Evil. This was not an easy switch: conf'ing Vi is a lot easier the conf'ing Emacs, as I'm not well versed with LISPs. I used @bbatsov's emacs-prelude to get up to speed, and get introduced to some "sane defaults" and "commonly used packages".

While it sometimes works, it also gave me a lot of headache as there is no version locking mechanism. So while today emacs-prelude (or any Emacs config w/o version locking) may work, tmrw may be broken.

I learned about `el-get`, how it allows version locking and how it groups all config surrounding a package together. Really nice in theory, but in practice my Emacs/Elisp skills were insufficient to get my own config going. I tried to base off @xaiohanyu's [oh-my-emacs](https://github.com/xiaohanyu/oh-my-emacs/), which uses both el-get and babel to achieve some solid and literal Emacs config, but I did not manage to port the configs from this repo to that set up.

Should I base of this repo so at least Haskell stuff works? Or put more effort into getting things to work in another setup? Or wait for the `haskell-ide-backend` stuff to land?

Any guidance is greatly appreciated... Thanks.

**Comment by xiaohanyu** [comment]
Actually, el-get has some version locking support, however, you need to add customized "recipes" to make it happen. I think a better way to do this is provide a binary distribution of `.emacs.d`, thus you just download a tarball, and extract, and it should work as expected.

**Comment by cies** [comment]
@xiaohanyu: Wouldn't a binary version make extending/personalizing is difficult?

**Comment by xiaohanyu** [comment]
No, I don't think so. You can still configure your emacs as you like, just in the normal way. Binary distribution `.emacs.d` is just like Ubuntu/Debian for linux kernel.
